dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdFileUploadTask.kt

```diff
@@ -0,0 +1,327 @@
+    var buildId: Provider<String> = providerFactory.provider { "" }
+
+    /**
+     * datadog-ci.json file, if found or applicable for the particular task.
+     */
+    @Optional
+    @get:InputFile
+    var datadogCiFile: File? = null
+
+    /**
+     * The sourceSet root folders.
+     */
+    @get:InputFiles
+    var sourceSetRoots: List<File> = emptyList()
+
+    /**
+     * The file containing the repository description.
+     */
+    @get:OutputFile
+    var repositoryFile: File = File("")
+
+    init {
+        group = DdAndroidGradlePlugin.DATADOG_TASK_GROUP
+        // it is never up-to-date, because request may fail
+        outputs.upToDateWhen { false }
+    }
+
+    @Internal
+    internal abstract fun getFilesList(): List<Uploader.UploadFileInfo>
+
+    /**
+     * Uploads the files retrieved from `getFilesList` to Datadog.
+     */
+    @TaskAction
+    @Suppress("TooGenericExceptionCaught")
+    fun applyTask() {
+        datadogCiFile?.let {
+            applyDatadogCiConfig(it)
+        }
+        applySiteFromEnvironment()
+        validateConfiguration()
+
+        check(!(apiKey.contains("\"") || apiKey.contains("'"))) {
+            INVALID_API_KEY_FORMAT_ERROR
+        }
+
+        check(buildId.isPresent && buildId.get().isNotEmpty()) {
+            MISSING_BUILD_ID_ERROR
+        }
+
+        val mappingFiles = getFilesList()
+        if (mappingFiles.isEmpty()) return
+
+        val repositories = repositoryDetector?.detectRepositories(
+            sourceSetRoots,
+            remoteRepositoryUrl
+        ).orEmpty()
+
+        if (repositories.isNotEmpty()) {
+            generateRepositoryFile(repositories)
+        }
+
+        val site = DatadogSite.valueOf(site)
+        var caughtErrors = mutableListOf<Exception>()
+        for (mappingFile in mappingFiles) {
+            LOGGER.info("Uploading ${mappingFile.fileType} file: ${mappingFile.file.absolutePath}")
+            try {
+                uploader.upload(
+                    site,
+                    mappingFile,
+                    if (repositories.isEmpty()) null else repositoryFile,
+                    apiKey,
+                    DdAppIdentifier(
+                        serviceName = serviceName,
+                        version = versionName,
+                        versionCode = versionCode,
+                        variant = variantName,
+                        buildId = buildId.get()
+                    ),
+                    repositories.firstOrNull(),
+                    !disableGzipOption.isPresent
+                )
+            } catch (e: Exception) {
+                caughtErrors.add(e)
+            }
+        }
+        // If any errors occurred, throw them as a single exception
+        if (caughtErrors.isNotEmpty()) {
+            if (caughtErrors.count() == 1) {
+                throw caughtErrors.first()
+            } else {
+                val consolidatedError = Exception("Multiple errors occurred during upload")
```

Comment:
Didn't notice it on the previous run, sorry. It is better to use `RuntimeException` here, normally nobody instantiates parent class `Exception` itself.

```suggestion
                val consolidatedError = RuntimeException("Multiple errors occurred during upload")
```